### ISSUE TITLE
Logout not working on Azure AD

### ISSUE THREAD

**Issue by TheL0w3R** [Clancey/simple_auth#76]
The Logout method is not deleting the cached token.
After running logout(), the custom chrome tab opens like the first time, but it instantly closes again, like the token never got deleted.
i've tried revoking the token by the AAD request, but when the tab opens again, it refreshes the token and logs me in again.

This is my first time writing a GitHub issue, sorry if my explanation is poor, i'm not english.

**Comment by Clancey** [comment]
This is outside my control. I cannot log someone out of chrome. I can only delete the user from inside the app.

**Comment by TheL0w3R** [comment]
I think I could help adding a way to complete the logout process, Azure offers an endpoint to invalidate the token, the problem is that it gets re-validated once you try to log in, as the user data is still cached in chrome.
The endpoint i'm talking about actually deletes the browser cache in order to complete the logout, but if you run it using the dart HTTP client, there'll be no cache to delete.

In other words, I'm not a Dart or Flutter expert, but I think the solution will be to open another custom tab with the logout URL and then return to the app.

Sorry if it bothers you, I just want to help :)

**Comment by Clancey** [comment]
Any of the api's can implement additional functionality on LogOut: https://github.com/Clancey/simple_auth/blob/master/simple_auth/lib/src/api/authenticatedApi.dart#L57

Just override that method. I am not aware of a general logout url, if there is one for azure that could be added in.

**Comment by ChiefWiggum** [comment]
I guess at least for Azure B2C the logout url is something like this:

`GET https://{tenant}.b2clogin.com/{tenant}.onmicrosoft.com/{policy}/oauth2/v2.0/logout?post_logout_redirect_uri={redirect_uri}`

For more B2C information have a look [here](https://docs.microsoft.com/en-us/azure/active-directory-b2c/active-directory-b2c-reference-oidc#send-a-sign-out-request).
